bundles/io/org.eclipse.smarthome.io.security/src/main/java/org/eclipse/smarthome/io/security/api/Cryptographic.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.security.api;
+
+import java.security.GeneralSecurityException;
+
+/**
+ * {@link Cryptographic} allows you to encrypt and decrypt. The algorithm and
+ * algorithm parameterisation is configured centralised for the framework.
+ *
+ * You can use the master password hash or a derived password to encrypt secrets, please be
+ * aware that a change of the master password can occur at any time though.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public interface Cryptographic {
```

Comment:
As @paulianttila [suggested](https://github.com/eclipse/smarthome/pull/4267/files#r139303413) this interface can be removed, because the `SecretsVaultService` is doing the encryption internally, right?

Reply:
Should we provide this anyway? We need something like this to implement the secret vault service and could export it for other use cases

Reply:
You mean for the secret vault service you need some methods that actually do the encryption and decryption of the configuration values? I am not sure if we should define an interface for that because this strongly depends on the implementation how this is done. As we might also need to provide an IV if the underlying cryptographic function needs it. So my suggestion is that we leave this entirely to the implementation of the secret vault service. @paulianttila  WDYT?